### PR TITLE
split easy controllers from giant start func

### PR DESCRIPTION
**Pull request by deads2k**
continues work from https://github.com/kubernetes/kubernetes/pull/37976 to separate controller initialization.

### CODE REVIEW

**Comment by deads2k** [comment]
Second commit.  It's code shuffling to shrink the init method and give a "normal" way to launch controllers.  This split will give us selective enablement.

File: cmd/kube-controller-manager/app/controllermanager.go [hunk 1]
```diff
@@ -252,99 +228,21 @@ func newControllerInitializers() map[string]InitFunc {
 	controllers["podgc"] = startEndpointController
```
Comment (sttts) [inline cmd/kube-controller-manager/app/controllermanager.go]: time for a renaming? podgc -> endpoint ?

File: cmd/kube-controller-manager/app/certificates.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
+/*
```
Comment (sttts) [inline cmd/kube-controller-manager/app/certificates.go]: Or even "groups" as in api-groups?

Comment (deads2k) [inline cmd/kube-controller-manager/app/certificates.go]: > Or even "groups" as in api-groups?

I wasn't trying to change up packages yet. I'm not even completely sure that I want to.  I was just trying to slim down the main type and get some consistency for startup.

File: cmd/kube-controller-manager/app/controllermanager.go (hunk 1 above)
Comment (deads2k) [inline cmd/kube-controller-manager/app/controllermanager.go]: > time for a renaming? podgc -> endpoint ?

typo.  I'm fixing it and a reversed condition now.

**Comment by deads2k** [comment]
@sttts let's tie this one off to avoid rebases.

**Comment by deads2k** [comment]
@k8s-bot cvm gce e2e test this

set up failure.

**Comment by deads2k** [comment]
@k8s-bot cvm gce e2e test this: issue #37241

File: cmd/kube-controller-manager/app/certificates.go (hunk 2 above)
Comment (sttts) [inline cmd/kube-controller-manager/app/certificates.go]: Oh, right. That's the `app` as in `cmd/*/app`. Though you are splitting up the controllers as apps.

**Pull request merged**